Tasks/AzureSpringCloudV0/deploymentProvider/AzureSpringCloudDeploymentProvider.ts

```diff
@@ -137,13 +136,24 @@ export class AzureSpringCloudDeploymentProvider {
                 } else {
                     throw Error(tl.loc('DeploymentDoesntExist', deploymentName));
                 }
-
             }
         }
+
+        // Determine the sku of the Azure Spring Cloud
+        const serviceSku = await this.azureSpringCloud.getServiceSku();
         try {
-            await this.azureSpringCloud.deploy(fileToUpload, sourceType, this.taskParameters.AppName,
-                deploymentName, createDeployment, this.taskParameters.RuntimeVersion, this.taskParameters.JvmOptions, 
-                this.taskParameters.EnvironmentVariables, this.taskParameters.DotNetCoreMainEntryPath, this.taskParameters.Version);
+            if (serviceSku == "S0") {
+                await this.azureSpringCloud.deploy(fileToUpload, sourceType, this.taskParameters.AppName,
+                    deploymentName, createDeployment, this.taskParameters.RuntimeVersion, this.taskParameters.JvmOptions, 
+                    this.taskParameters.EnvironmentVariables, this.taskParameters.DotNetCoreMainEntryPath, this.taskParameters.Version);
+            } else if (serviceSku == "E0") {
+                await this.azureSpringCloud.deployWithBuildService(fileToUpload, sourceType, this.taskParameters.AppName, 
+                    deploymentName, createDeployment, this.taskParameters.RuntimeVersion, this.taskParameters.JvmOptions,
+                    this.taskParameters.EnvironmentVariables, this.taskParameters.DotNetCoreMainEntryPath, this.taskParameters.Version, this.taskParameters.Builder);
+            } else {
```

Comment:
We have Basic tier as B0. I think you can handle basic tier in the same way as standard tier